src/components/motor/MotorController.cpp

```diff
@@ -1,43 +1,40 @@
 #include "components/motor/MotorController.h"
 #include <hal/nrf_gpio.h>
 #include "systemtask/SystemTask.h"
-#include "app_timer.h"
 #include "drivers/PinMap.h"
 
-APP_TIMER_DEF(shortVibTimer);
-APP_TIMER_DEF(longVibTimer);
-
 using namespace Pinetime::Controllers;
 
 void MotorController::Init() {
   nrf_gpio_cfg_output(PinMap::Motor);
   nrf_gpio_pin_set(PinMap::Motor);
-  app_timer_init();
 
-  app_timer_create(&shortVibTimer, APP_TIMER_MODE_SINGLE_SHOT, StopMotor);
-  app_timer_create(&longVibTimer, APP_TIMER_MODE_REPEATED, Ring);
+  shortVib = xTimerCreate("shortVib", 1, pdFALSE, nullptr, StopMotor);
+  longVib = xTimerCreate("longVib", 1, pdTRUE, this, Ring);
 }
 
-void MotorController::Ring(void* p_context) {
-  auto* motorController = static_cast<MotorController*>(p_context);
+void MotorController::Ring(TimerHandle_t xTimer) {
+  auto motorController = static_cast<MotorController*>(pvTimerGetTimerID(xTimer));
+  xTimerChangePeriod(motorController->longVib, pdMS_TO_TICKS(1000), 0);
```

Comment:
`longVib` can be initialized to the correct period and then changing the period afterwards can be avoided. `StartRinging()` needs to be tweaked slightly

Reply:
I can make that change, but that would delay by one second the start of the vibration.
An alternative could be starting the two timers at the same time, but that didn't work in my previous testing, I think it's because of unintended execution order when you start two timers simultaneously.

Reply:
The two timers should be started simultaneously like before. It seems to work fine for me. `Ring(this)` can't be used, because the function now expects a `TimerHandle_t` instead of a simple pointer. Just duplicating the RunForDuration(50) should be good enough for now, or alternatively a new function in an unnamed namespace can be created.

Reply:
I see, can you push the modification that you tested?